### PR TITLE
More tidying of code and memory leaks

### PR DESCRIPTION
**Pull request by oliveromahony**
### Proposed changes

Removed some memory leaks and used more performant functions

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [x] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: src/core/environment.go [hunk 1]
```diff
@@ -540,7 +543,7 @@ func (env *EnvironmentType) Processes() (result []*Process) {
 
 		newProcess := &Process{
 			Pid:        pid,
-			Name:       name,
+			Name:       string([]byte(name)),
```
Comment (dhurley) [inline src/core/environment.go:546]: Not sure why this change is needed?

**Review by aphralG — APPROVED** [review]

File: src/core/environment.go (hunk 1 above)
Comment (oliveromahony) [reply src/core/environment.go:546]: this makes a copy of the name and puts it in a new variable. I'll remove it

**Review by dhurley — APPROVED** [review]
